ng2-material/components/data_table/data_table.ts

```diff
@@ -0,0 +1,199 @@
+import {Component, Input, Output, Inject, forwardRef, ElementRef, AfterContentInit, EventEmitter} from 'angular2/core';
+import {Observable} from 'rxjs/Observable';
+import {Observer} from 'rxjs/Observer';
+import 'rxjs/add/operator/filter';
+import 'rxjs/add/operator/share';
+import {MdCheckbox} from '../checkbox/checkbox';
+
+/**
+ * @name mdDataTable
+ *
+ * @description
+ * The `<md-data-table>` component is an enhanced version of data table
+ *
+ * Enhancement:
+ *   - selectable row (in progress)
+ *   - sortable column (@todo)
+ *
+ * It renders rows and cells as they're set by the user so it doesn't manage the inline editing by itself.
+ *
+ * Available attributes are:
+ *  - `selectable` - (boolean) Enable one checkbox per line and a master checkbox to rule them all.
+ *  - `selectableValue` on tr markup- (string) value of the checkbox. If it's not set the checkbox's value will be the index of the row.
+ *  - `truncateHeaderLabelAt` - (int) Truncate too long header label and prepare a tooltip for the rest of the label. (@todo)
+ *
+ * Available classes are:
+ *  - `th.md-sortable` - Enable the order of a column. (@todo)
+ *  - `td.md-data-table__cell--non-numeric` - Disable the right alignement of the cells.
+ *
+ * Available output is:
+ *  - `onSelectableChange` - fired when a checkbox change in the MdDataTable. It's an object which contain.
+ *  * - `name`      - selectable_all | selectable_change
+ *    - `taget`     - the MdDataTable
+ *    - `values`    - array of selected values
+ *
+ * @usage
+ * ### Default table
+ * <hljs>
+ * <md-data-table>
+ *     <thead>
+ *         <tr>
+ *             <th class="md-data-table__cell--non-numeric">Material</th>
+ *             <th>Quantity</th>
+ *             <th>Unit price</th>
+ *         </tr>
+ *     </thead>
+ *     <tbody>
+ *         <tr>
+ *             <td class="md-data-table__cell--non-numeric">Acrylic (Transparent)</td>
+ *             <td>25</td>
+ *             <td>$2.90</td>
+ *         </tr>
+ *     </tbody>
+ * </md-data-table>
+ *
+ * ### Default table with selectable rows
+ * <md-data-table [selectable]="true">
+ *     <thead>
+ *         <tr>
+ *             <th class="md-data-table__cell--non-numeric">Material</th>
+ *             <th>Quantity</th>
+ *             <th>Unit price</th>
+ *         </tr>
+ *     </thead>
+ *     <tbody>
+ *         <tr [selectableValue]="'42'">
+ *             <td class="md-data-table__cell--non-numeric">Acrylic (Transparent)</td>
```

Comment:
I'm not 100% sure about this, but I think it would be more inline with the existing components to use hyphens to separate each component.

`.md-data-table-cell` `.md-data-table-cell-non-numeric`, etc.